src/modules/user/infraestructure/persistence/user.mysql.repository.ts

```diff
@@ -0,0 +1,89 @@
+import { Body } from '@nestjs/common';
+import { CreateUserDto, UpdateUserDto } from '../../application/dtos';
+import type { Repository } from 'typeorm';
+import {
+  checkIfUserAlreadyExists,
+  checkIfUserExists,
+  checkIfIsCreateBodyValid,
+  checkIfIsUpdateBodyValid,
+} from '../utils/user-validations';
+import { throwRepositoryException } from '../utils/repository-exceptions';
+import { RepositoryErrorsMessages } from '../errors/repository-errors';
+import { InjectRepository } from '@nestjs/typeorm';
+import { User } from '../../domain/models/user.entity';
+
+export class UserMySqlRepository {
+  constructor(
+    @InjectRepository(User)
+    private readonly userRepository: Repository<User>,
+  ) {}
+
+  async createUser(@Body() user: CreateUserDto) {
+    checkIfIsCreateBodyValid(user.nickName, user.email, user.password);
```

Comment:
You can use `class-validator` in your dtos and catch these errors in your outer layer.